specification.md

```diff
@@ -393,20 +393,29 @@ See paragraph 5.1 of [WCA Competition Requirements Policy](https://www.worldcube
 
 | Attribute | Type | Description |
 | --- | --- | --- |
-| `when` | [`DateTime`](#datetime) | The point in time at which the qualification requirement must be satisfied. |
-| `type` | `"single"\|"average"` | The type of result the requirement refers to. |
-| `attemptResult` | [`AttemptResult`](#attemptresult) | The attempt result that needs to be beaten in order to qualify to register for the event. |
+| `whenDate` | [`Date`](#date) | The date by which the qualification requirement must be satisfied.  If a result is set in a multiple-day competition which begins by this date, that is considered to have been set by this date. |
+| `type` | `"single"\|"average"\|"ranking"` | The type of result the requirement refers to. |
+| `attemptResult` | [`AttemptResult`](#attemptresult)\|`null` | The attempt result that needs to be beaten in order to qualify to register for the event. Required if `type` is "single" or "average". |
+| `ranking` | `Integer\|null` | The number of competitors to qualify for this event. Required if `type` is "ranking". |
```

Comment:
Since `attemptResult` and `ranking` are exclusive, I think we should go the same route as with `AdvancementCondition` for consistency, so we would have `type` and `level`.

Reply:
Good idea, done.